src/cluster_legacy.c

```diff
@@ -110,6 +110,31 @@ static void clusterBuildMessageHdr(clusterMsg *hdr, int type, size_t msglen);
 void freeClusterLink(clusterLink *link);
 int verifyClusterNodeId(const char *name, int length);
 
+int isClusterSlotsResponseCached(connTypeForCaching conn_type) {
+    if (server.cluster->cached_cluster_slot_info[conn_type] &&
+        sdslen(server.cluster->cached_cluster_slot_info[conn_type])) {
```

Comment:
As we discussed, it's weird that we have a check for zero length string here. We will leak the memory later when we call cacheSlotsResponse() if this were ever true. As far as I know, the cached response should never be zero length.